api/models/Notification.js

```diff
@@ -86,6 +94,28 @@ module.exports = bookshelf.Model.extend({
     })
   },
 
+  sendJoinRequestPush: function () {
+    var communityIds = Activity.communityIds(this.relations.activity)
+    if (isEmpty(communityIds)) return Promise.resolve()
```

Comment:
we should probably throw an error here and in `sendApprovedJoinRequestPush` instead of doing nothing -- [fail fast](https://en.wikipedia.org/wiki/Fail-fast)